spec/scenarios/dunning/campaign_v1_spec.rb

```diff
@@ -0,0 +1,232 @@
+  let(:dunning_campaign_threshold) do
+    create(:dunning_campaign_threshold, dunning_campaign:, amount_cents: 150_00, currency: "EUR")
+  end
+  let(:stripe_cus_id) { "cus_123456789" }
+  let(:stripe_pm_id) { "pm_123456" }
+
+  let(:stripe_provider) { create(:stripe_provider, organization:) }
+  let(:customer) { create(:customer, organization:, payment_provider: :stripe, payment_provider_code: stripe_provider.code, net_payment_term: 2) }
+  let(:stripe_customer) { create(:stripe_customer, customer:, payment_provider: stripe_provider, provider_customer_id: stripe_cus_id) }
+
+  let(:external_subscription_id) { "sub_overdue-dunning-campaign-v1" }
+  let(:plan) { create(:plan, organization:, pay_in_advance: true, amount_cents: 149_00) }
+  let!(:addon) { create(:add_on, organization:) }
+
+  let(:webhooks_sent) { [] }
+
+  let(:stripe_customer_response) do
+    File.read("spec/fixtures/stripe/customer_retrieve_response.json")
+  end
+  # TODO: this is part of another PR coming soon https://github.com/getlago/lago-api/pull/3345
+  let(:stripe_payment_method_response) do
+    {
+      id: stripe_pm_id,
+      object: "payment_method",
+      allow_redisplay: "always",
+      billing_details: {
+        address: {
+          city: nil,
+          country: "FR",
+          line1: nil,
+          line2: nil,
+          postal_code: nil,
+          state: nil
+        },
+        email: "awdawd@desf.com",
+        name: "Testing Stripe",
+        phone: nil
+      },
+      card: {
+        brand: "visa",
+        checks: {
+          address_line1_check: nil,
+          address_postal_code_check: nil,
+          cvc_check: "pass"
+        },
+        country: "US",
+        display_brand: "visa",
+        exp_month: 12,
+        exp_year: 2028,
+        fingerprint: "8TOiB4cGytYxCweY",
+        funding: "credit",
+        generated_from: nil,
+        last4: "4242",
+        networks: {
+          available: [
+            "visa"
+          ],
+          preferred: nil
+        },
+        regulated_status: "unregulated",
+        three_d_secure_usage: {
+          supported: true
+        },
+        wallet: nil
+      },
+      created: 1741878064,
+      customer: stripe_cus_id,
+      livemode: false,
+      metadata: {},
+      type: "card"
+    }
+  end
+  let(:stripe_payment_intent_response) do
+    File.read("spec/fixtures/stripe/payment_intent_failed_card_declined.json")
+  end
+
+  def build_stripe_payment_intent_response
+    JSON.parse(stripe_payment_intent_response, symbolize_names: true).tap do |h|
+      h[:error][:payment_intent][:id] = "pi_#{SecureRandom.hex}"
+    end
+  end
+
+  # TODO: make it a test metadata `:scenarios, type: :request, premium: true`
+  around { |test| lago_premium!(&test) }
+
+  before do
+    stub_pdf_generation
+    stripe_customer
+    dunning_campaign_threshold
+
+    stub_request(:post, webhook_url).with do |req|
+      webhooks_sent << JSON.parse(req.body)
+      true
+    end.and_return(status: 200)
```

Comment:
We log Lago webhook sent so we can verify all the correct webhook were sent.
If we like it, this should be refactored to the ScenarioHelper to be reused.

**Example of expectations**
```ruby
expect(webhooks_sent.map { _1["webhook_type"] }).to eq(%w[
  subscription.started
  invoice.created
  invoice.generated
  invoice.payment_failure
])
```